coordinator/src/services/translation/organization.ts

```diff
@@ -1,48 +1,125 @@
 import * as fhir from "../../model/fhir-resources"
 import * as peoplePlaces from "../../model/hl7-v3-people-places"
-import {getCodeableConceptCodingForSystem, getIdentifierValueForSystem, onlyElement, resolveReference} from "./common"
+import {
+  getCodeableConceptCodingForSystemOrNull,
+  getIdentifierValueForSystem, getResourceForFullUrl,
+  onlyElement,
+  resolveReference
+} from "./common"
 import * as codes from "../../model/hl7-v3-datatypes-codes"
 import * as core from "../../model/hl7-v3-datatypes-core"
 import {convertAddress, convertTelecom} from "./demographics"
 
-function convertOrganization(fhirOrganization: fhir.Organization) {
-  const hl7V3Organization = new peoplePlaces.Organization()
+const NHS_TRUST_CODE = "RO197"
 
-  const organizationSdsId = getIdentifierValueForSystem(fhirOrganization.identifier, "https://fhir.nhs.uk/Id/ods-organization-code")
-  hl7V3Organization.id = new codes.SdsOrganizationIdentifier(organizationSdsId)
+/**
+ * TODO - This mapping is a temporary measure for testing. We're reasonably confident that it's correct for primary
+ * care prescriptions, but we've not yet agreed where the two organizations should come from for secondary care
+ * prescriptions.
+ * TODO - possible bug, does current implementation depend on ordering of Organizations in FHIR message?
+ */
+export function convertOrganizationAndProviderLicense(
+  fhirBundle: fhir.Bundle,
+  fhirOrganization: fhir.Organization,
+  fhirHealthcareService: fhir.HealthcareService
+): peoplePlaces.Organization {
+  const hl7V3Organization = convertRepresentedOrganization(fhirOrganization, fhirHealthcareService, fhirBundle)
+
+  hl7V3Organization.healthCareProviderLicense = convertHealthCareProviderLicense(fhirOrganization, fhirBundle)
+
+  return hl7V3Organization
+}
 
-  if (fhirOrganization.type !== undefined) {
-    const organizationTypeCoding = getCodeableConceptCodingForSystem(fhirOrganization.type, "https://fhir.nhs.uk/R4/CodeSystem/organisation-type")
-    hl7V3Organization.code = new codes.OrganizationTypeCode(organizationTypeCoding.code)
+function convertRepresentedOrganization(fhirOrganization: fhir.Organization, fhirHealthcareService: fhir.HealthcareService, fhirBundle: fhir.Bundle) {
+  const organizationTypeCoding = getCodeableConceptCodingForSystemOrNull(fhirOrganization.type, "https://fhir.nhs.uk/R4/CodeSystem/organisation-role")
+  const representedOrganization = (organizationTypeCoding?.code === NHS_TRUST_CODE) ? new CostCentreHealthcareService(fhirHealthcareService, fhirBundle) : new CostCentreOrganization(fhirOrganization)
```

Comment:
Change the condition to be `!organizationTypeCoding || organizationTypeCoding.code === NHS_TRUST_CODE`. See my comment on `resolveOrganizationReference()`.